carryall/go-google-scraper-challenge
# [Chore-1] Setup database and ORM

## What happened

Resolved #17 

- use docker compose to define database service for development environment

## Insight

N/A

## Proof Of Work

Tested locally by
- run `bee generate migration user` to generate a test migration
- add the following SQL to the migration file to create a table
- 
```Go
func (m *User_20210111_120028) Up() {
	// use m.SQL("CREATE TABLE ...") to make schema update
	m.SQL(`CREATE TABLE "user"
	(
		id SERIAL,
		email text UNIQUE,
		created_at timestamp NOT NULL,
		updated_at timestamp NOT NULL,
		CONSTRAINT user_pkey PRIMARY KEY (id)
	);`)
}
```

- run `make dev`

- the migration should be successfully run
![Screen Shot 2564-01-11 at 12 47 34](https://user-images.githubusercontent.com/1772999/104150075-4097dd80-540b-11eb-99ee-163c81a79a22.png)

- the table should be created on the Database
![Screen Shot 2564-01-11 at 12 48 38](https://user-images.githubusercontent.com/1772999/104150106-5a392500-540b-11eb-9dc9-4a27f89ebd2a.png)

- then generate a `User` model with this command
```
bee generate model user -fields="email:string"
```

- create a user
```Go
userId, err := models.AddUser(&models.User{Email: "john@nimbhehq.co"})
fmt.Printf("ID: %d, ERR: %v\n", userId, err)
```
- the database should have the created record
![Screen Shot 2564-01-11 at 13 14 03](https://user-images.githubusercontent.com/1772999/104151287-3546b100-540f-11eb-934a-41876d8ed4b9.png)
